src/vm.rs

```diff
@@ -515,20 +516,31 @@ where
     }
 
     fn value_to_cloak_value(&mut self, value: &Value) -> spacesuit::AllocatedValue {
+        let assignment = match self.variable_assignment(value.qty) {
```

Comment:
It's safer and easier to read to match on the entire tuple:

```rust
let assignment = match (self.variable_assignment(value.qty), self.variable_assignment(value.flv)) {
    (Some(ScalarKind::Integer(q)), Some(ScalarKind::Scalar(f))) => spacesuit::Value { q, f },
    (_, _) => None
}
```

Reply:
Actually, `None` should return on `(None, None)` only, everything else is an inconsistent input and must yield an error like `VMError::InconsistentInput` or something like that.